waymo2argo/waymo_raw_data_to_argoverse.py

```diff
@@ -0,0 +1,441 @@
+#!/usr/bin/env python3
+
+import argparse
+import glob
+import imageio
+import itertools
+import json
+import math
+import numpy as np
+import os
+import pandas as pd
+from pathlib import Path
+import pdb
+from typing import Any, Dict, List, Union
+import uuid
+
+import cv2
+from pyntcloud import PyntCloud
+from scipy.spatial.transform import Rotation
+import tensorflow.compat.v1 as tf
+
+tf.enable_eager_execution()
+
+from argoverse.utils.json_utils import save_json_dict
+from argoverse.utils.se3 import SE3
+import waymo_open_dataset
+from waymo_open_dataset.utils import range_image_utils
+from waymo_open_dataset.utils import transform_utils
+from waymo_open_dataset.utils import frame_utils
+from waymo_open_dataset import dataset_pb2 as open_dataset
+
+from waymo2argo.transform_utils import (
+    rotX,
+    rotY,
+    rotmat2quat,
+    quat2rotmat,
+    yaw_to_quaternion3d,
+)
+
+"""
+Extract poses, images, and camera calibration from raw Waymo Open Dataset TFRecords.
+
+See the Frame structure here:
+https://github.com/waymo-research/waymo-open-dataset/blob/master/waymo_open_dataset/dataset.proto
+
+See paper:
+https://arxiv.org/pdf/1912.04838.pdf
+"""
+
+# Mapping from Argo Camera names to Waymo Camera names
+# The indices correspond to Waymo's cameras
+CAMERA_NAMES = [
+    "unknown",  # 0, 'UNKNOWN',
+    "ring_front_center",  # 1, 'FRONT'
+    "ring_front_left",  # 2, 'FRONT_LEFT',
+    "ring_front_right",  # 3, 'FRONT_RIGHT',
+    "ring_side_left",  # 4, 'SIDE_LEFT',
+    "ring_side_right",  # 5, 'SIDE_RIGHT'
+]
+
+# Mapping from Argo Label types to Waymo Label types
+# The indices correspond to Waymo's label types
+LABEL_TYPES = [
+    "OTHER_MOVER",  # 0, TYPE_UNKNOWN
```

Comment:
I'm still a bit confused about this part.

Maybe we can add one more inline comment saying Waymo labels are on the right, and
corresponding Argoverse labels are on the left, with a link to https://github.com/argoai/argoverse-api/blob/master/argoverse/data_loading/object_classes.py#L6

Or are there just 5 label types in Waymo Open Dataset?

I'm not sure that Waymo's `UNKNOWN` would correspond to Argoverse' `OTHER_MOVER`?

Reply:
Addressed [here](https://github.com/johnwlambert/waymo_to_argoverse/pull/2/commits/4ce61dbfa76327be7e9627546897df849c7b8cff)

Yeah according to their GitHub, there are 5 label types.